packages/playwright/src/transform/compilationCache.ts

```diff
@@ -218,9 +218,15 @@ export function collectAffectedTestFiles(dependency: string, testFileCollector:
     if (deps.has(dependency))
       testFileCollector.add(testFile);
   }
-  for (const [testFile, deps] of externalDependencies) {
-    if (deps.has(dependency))
-      testFileCollector.add(testFile);
+  for (const [importingFile, deps] of externalDependencies) {
+    if (deps.has(dependency)) {
+      testFileCollector.add(importingFile);
```

Comment:
We should only add `importingFile` if it is a test file. That's what the line 215 checks for.

Reply:
This is not a performance-critical place, so premature optimization is not needed. Overall, we are serious about performance on the critical path, and in general in the core.